### ISSUE TITLE
[Feature suggestion ] Add donate/sponsor notification when launch spyder IDE 

### ISSUE THREAD

**Issue by Po-Hsuan-Huang** [spyder-ide/spyder#7437]
Spyder lacks proper funding and most users aren't aware of this. 

https://opencollective.com/spyder#updates

Please add a donate/sponsor notification when launch Spyder IDE so people know it is underfunded.

**Comment by ccordoba12** [comment]
Instead of showing a message every time Spyder starts we could show one in the *About Spyder* dialog.

@CAM-Gerlach, what do you think?

**Comment by CAM-Gerlach** [comment]
Its a good idea to have it there as well for completeness, but in order to see it, users would essentially have to almost already be looking for it. It seems unlikely a user would open the About dialog except to look for some specific information there (mostly version information), and ignore the rest; furthermore, those users are likely to mostly be the small percentage who would already see our message on Github anyway since typically they would do so to report a bug/problem or otherwise interested in finding out who we are.

Here's an idea I really like that might help us (and our users) with multiple problems at once: A proper welcome screen, like what [ATOM has](https://discuss.atom.io/t/how-do-i-make-the-welcome-screen-stop-showing-up/40403). It would be a non-model dialog/window with a one-line description of Spyder and the project's community/volunteer/nonprofit status, links to our website and various resources for getting started (e.g. the tour, the tutorial), getting help (our docs, the Troubleshooting Guide), getting in touch (Google Groups, Gitter) and (prominently at the end) getting involved (Github, OpenCollective). It would then have a checkbox that says "Don't show this again" which would persist indefinitely (unless the Spyder preferences were deleted/reset).

We can also have a similar dialog that appears once on the first startup of a new version of Spyder, that welcomes the user to the new version, and in 2-3 sentences summarizes the key new features. It would also have a link to the release post on the blog (essentially an accessible, "visual" changelog with the key points about the new version) and the full changelog, and a highly abbreviated form of the resources section from the main welcome screen (with just Docs, Troubleshooting Guide, Google Groups, and Gitter inline links). At the bottom, similar to before, we'd have the support us blurb. We could also use that space in the future to thank contributors or backers, e.g. as part of an OC tier (which we really should set up sometime soon; I've already got a bunch of ideas for levels and rewards on the drive, none of which really cost us anything).

Some example text for the blurb:

Spyder is entirely community-funded and developed by Python programmers and data scientists just like you. If you'd like to help us add continued enhancements and new features, please support us with a donation or monthly pledge [on OpenCollective](link), or contribute your own changes to our pure Python codebase [on Github](link).

**Comment by Po-Hsuan-Huang** [comment]
Hi, 
I just started to pay monthly, and I think it will be nice to have a progress bar on welcome screen about how much have been raised and how far it is from the fundraising goal, just like how Wikipedia did, and advertise a little bit what will users get if the goal is accomplished.

**Comment by CAM-Gerlach** [comment]
@Po-Hsuan-Huang I definitely agree on having a public goal, and in fact advocated for one previously, but there was no clear consensus at the time. However, as you describe it serves to both motivate potential backers and be transparent about our objectives and progress to existing ones, and if we set one on OC it will show something very similar to what you're requesting (though I'm not sure about putting it directly in the welcome screen; we probably don't want to focus it too explicitly on getting donations). 

Coupled with additional OC tiers and "rewards" (basically thank yous/backers' names various places) as I've also proposed before, which are very simple to do and cost us nothing), as well as the welcome screen, this could really help us get on a more sustainable financial footing, so I again recommend going forward with it if others agree, and I'd be the one doing most if not all of the modest amount of work to implement it.

**Comment by Po-Hsuan-Huang** [comment]
@CAM-Gerlach  I understand and respect all your decisions. As a backer, it is not because I am charitable or anything better, but if the issues of users can't be solved timely, Spyder will loose users. That's a potential problem that might be overlooked. Being free is not always the most cheapest decision to users. To be honest, I might switch to other platforms that solve my problems even though I would have to pay more. The fundraising goal is not even a big number, my naive hope is that a sprint fundraising campaign will help we get over with it instead of prolonged agony.

**Comment by ccordoba12** [comment]
> Here's an idea I really like that might help us (and our users) with multiple problems at once: A proper welcome screen

Good idea! Do you want to work on it for Spyder 4 @CAM-Gerlach?

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Sure! It shouldn't be too hard, even for me :)

**Comment by ccordoba12** [comment]
I don't think so. But you need to use a WebView widget instead of regular CodeEditor to show the welcome screen on the Editor. Or we could show the welcome message in the Help pane, which should be far more easier to implement.

**Comment by CAM-Gerlach** [comment]
I was thinking of using a fancy non-modal dialog instead of embedding it in an existing panel, which would make it hard to miss, easy to implement and simple to have a "Show welcome screen on startup" checkbox option. Furthermore, it would allow us to (eventually) implement further functionality in the future like many other IDEs have, particularly "jumping off points" like buttons for creating new projects (which could integrate the cookiecutter template functionality planned for Spyder 4), and a list of recent files/projects to open and get started. Then it can become something much more useful to users than just a nag screen, to the point where many might like to leave it open full time.

However, if you really prefer I integrate it into an existing panel, then the Help pane would definitely seem the most natural other choice, since its way easier than in the Editor :) and it also fits conceptually with it, it already has what is sort of a very rudimentary welcome screen. Also, the main advantage to it over the dialog would be that we could simply always open it on startup with no need for an option, since it doesn't obstruct anything and there isn't anything else that would otherwise be displayed there. However, the flipside of that is that it goes away, never to return, as soon as the user hits the inspect shortcut or (if connections are enabled) types a callable, which could be undesired. It is also less visible and conventional than a dialog, and most importantly doesn't easily allow (to my knowledge) for the same kind of dynamic functionally mentioned above that a dialog would.

Why not take the best of both worlds: I can implement the proper welcome screen as a dialog as discussed here, but also improve the existing Help pane "Usage" screen that displays on startup to modestly expand and update the text, and display some additional links to useful help resources besides just the tutorial. This gives us the advantages of both approaches, with very little additional work.
